pkg/apis/core/helper/helpers.go

```diff
@@ -512,32 +512,44 @@ func DeterminePrimaryIPFamily(ipFamilies []core.IPFamily) core.IPFamily {
 	return ipFamilies[0]
 }
 
-// KubeAPIServerFeatureGateEnabled returns whether the given feature gate is enabled for the kube-apiserver for the given Shoot spec.
-func KubeAPIServerFeatureGateEnabled(shoot *core.Shoot, featureGate string) bool {
+// KubeAPIServerFeatureGateDisabled returns whether the given feature gate is explicitly disabled for the kube-apiserver for the given Shoot spec.
+func KubeAPIServerFeatureGateDisabled(shoot *core.Shoot, featureGate string) bool {
```

Comment:
Why did you reverse the function names and their return values instead of reversing the invocation part only? I think generally we try to use "enabled" for more straight-forward readability/usability, see also our APIs. Revert it?

Reply:
If we just reverse the outpout of `KubeAPIServerFeatureGateEnabled`, then for `kubeAPIServer == nil` we get true, but we actually want false. Similarly, when the feature gate does not exist in the map, we need false.

Reply:
What? 😂

Reply:
`KubeAPIServerFeatureGateEnabled ` returns false when `kubeAPIServer == nil`.
Then `! KubeAPIServerFeatureGateEnabled` is true.
When checking if the feature gate is "explicitly" disabled (since it is enabled by kubernetes by default), `kubeAPIServer == nil` should return false.